eZ/Publish/Core/Search/Common/Indexer.php

```diff
@@ -46,16 +47,23 @@
      */
     protected $searchHandler;
 
+    /**
+     * @var \eZ\Publish\Core\Search\Common\IndexerErrorHandler
+     */
+    protected $errorHandler;
+
     public function __construct(
         LoggerInterface $logger,
         PersistenceHandler $persistenceHandler,
         DatabaseHandler $databaseHandler,
-        SearchHandler $searchHandler
+        SearchHandler $searchHandler,
+        IndexerErrorHandler $errorCollector = null
     ) {
         $this->logger = $logger;
         $this->persistenceHandler = $persistenceHandler;
         $this->databaseHandler = $databaseHandler;
         $this->searchHandler = $searchHandler;
+        $this->errorHandler = $errorCollector ?: new NullErrorHandler();
```

Comment:
handler or collector? lets pick one, Collector might be the clearest word here for what it does.

wdyt @alongosz ?

Reply:
> handler or collector? lets pick one, Collector might be the clearest word here for what it does.

Well... the interface has the `handle` method, so I'd say handler. But AFAICS `handle` only collects these errors, right? In that case it should be called `collect` and the interface and implementing classes - `*Collector`

Reply:
Renamed to `Collector` in https://github.com/ezsystems/ezpublish-kernel/pull/2221/commits/4f8d3f0543ea1ddaae33c42a945ab2cc63c2def2.